src/plugin/advancedCompare/index.js

```diff
@@ -0,0 +1,42 @@
+import * as C from '../../constant'
+
+export default (option, dayjsClass) => {
+  const proto = dayjsClass.prototype
+  const oldSame = proto.isSame
+  const oldIsAfter = proto.isAfter
+  const oldIsBefore = proto.isBefore
+
+  const v = (l, r) => l.isValid() && r.isValid()
+
+  dayjsClass.prototype.isSame = function (o, u) {
+    if (!v(this, o)) return false
+
+    return (u && this.$utils().prettyUnit(u) === C.MS)
+      ? oldSame.bind(this)(o)
+      : this.startOf(u).valueOf() <= o && o <= this.endOf(u).valueOf()
+  }
+
+  dayjsClass.prototype.isAfter = function (o, u) {
+    if (!v(this, o)) return false
+
+    return (u && this.$utils().prettyUnit(u) === C.MS)
+      ? oldIsAfter.bind(this)(o)
+      : o.valueOf() < this.startOf(u).valueOf()
+  }
+
+  dayjsClass.prototype.isBefore = function (o, u) {
+    if (!v(this, o)) return false
+
+    return (u && this.$utils().prettyUnit(u) === C.MS)
```

Comment:
and seems we don't need `u &&` scince prettyUnit has already done this check

Reply:
You are right, that check was redundant. Fixed with bba77e4e246effc0070fb5fb6ae4f8b8f8e30f96.